### DISCUSSION TITLE
Sidecar Installer

### DISCUSSION THREAD

**Discussion by kieranmetcalfe** [airsdk/Adobe-Runtime-Support#1067] [General]
Hi,

In the past, we created Sidecar installer packages for our AIR App. With the new Harman version, the old ones no longer function (possibly expired certs from the Adobe installer).

Is it possible to use the Harman installer to create updated sidecars, and what licensing steps are needed?

Thanks,

Kieran

**Comment by ajwfrost** [comment]
Yes I imagine that the certificate differences would cause a problem if you're trying to run an installer for an old (Adobe-published) version of AIR on top of our runtime..

Can I ask why you're using a sidecar mechanism i.e. installing both the shared runtime and your application, within the same package? It would seem better to create a 'bundle' which has your application with the captive runtime embedded in it, and then you're distributing something that is self-contained and unaffected by other installations on the machine?

Could you please also clarify what platform(s) you're targeting and in particular if it's Mac, what variants of the OS and processors that you're looking at? There are extra challenges with the shared runtime on recent macOS versions where it's enforcing code-signing on running applications (since, there's no code-signing when an .air package is used with the shared runtime...)

thanks

**Comment by kieranmetcalfe** [comment]
Hi! I didn't appear to get a notification of the reply (18 months ago) and the project was quite low priority, but it is now raising its head again... 

So - we used a sidecar setup (on the assumption I have the right terminology) as it was what Adobe licensed at the time - the original version of the app was about 2013. 

Can you detail the difference between sidecar and bundle? according to https://help.adobe.com/en_US/air/redist/WS485a42d56cd19641-70d979a8124ef20a34b-8000.html they were the same thing: 

"Bundled deployment of Adobe AIR Runtime Installer files—this is also known as a “sidecar installation.” "

So, the endgame for us (and the reason for the renewed interest) would be to bundle / sidecar the app with the older WebKit runtimes from https://airsdk.harman.com/runtime. Is that doable - does it work the same way as the adobe one on the above link, and what licensing steps would be involved?

**Reply by ajwfrost** [reply]
When it talks about a sidecar installation, it means there is an installer for the shared AIR runtime (which normally you could install from our website under the 'runtime' area) which is bundled into your own application's installer.

When we talk about a 'bundle' though, we mean a version of your application that includes its own (private) copy of the runtime files that are needed here. So if you use the AIR SDK and create a 'bundle' then you end up with an application folder which contains all that you need; you can then use a standard installation application to create an installer from that (msi/pkg/etc)

If you're needing the older webkit runtimes, then you really don't want to do the sidecar thing, because this would mean it ships with 33.1.1.744 of the shared AIR runtime, and if you install your app on any computer that already has a later version of AIR - or if anyone installs the shared AIR runtime with a later version - then your app wouldn't be usable. For this scenario, it's pretty essential to be using the AIR SDK to create a bundle, and then turning that into an installer.

From the licensing perspective, this simplifies things a little: the AIR SDK license would be applicable, based on your company category/revenue per https://airsdk.harman.com/pricing, and this license then gives you the rights to redistribute the AIR runtime files when they're packaged up with your application.

Can I check what operating system(s) you need to support? Windows is a bit easier to create the bundle, but it's not easy to create an MSI from this without third-party tools; macOS conversely has got the pkgbuild/productbuild tools to create PKG files but is a bit of a pain for signing and notarization..

thanks

**Reply by kieranmetcalfe** [reply]
Thanks for the reply - that's helpful stuff. That's a good point about the versions etc especially.

However, at present, our user base is quite small, and the owners are non-profit/charity without the funds for too much work on this - we're currently in the process of porting the entire app to a web-based platform (for various reasons, but the fact that the Adobe Air Development software became unusable for a few years forced our hand, sadly).

At this point, we're essentially limping the app along until we can officially retire it later this year. So, it does seem that that the sidecar approach will meet our very short-term needs. If a user has a newer Air running, then our support team can just send them air file (and they'll have to live with some error messages that get thrown as a result).

It's certainly not ideal, but we don't have the funds to get new code certs to make sweeping changes for the app, so we're just supporting the existing AIR package file as best we can :)

I've had a look and seem to have been able to create OSX DMG and Windows Exe sidecars from the relevant Runtime files with our AIR packages, so I think we just need to get an agreement in place to distribute that. Would that be covered on the Free plan? We're happy with the splash screen for all the reasons above.

**Reply by kieranmetcalfe** [reply]
(If it's not covered, we'll just link to the runtime and .air as separate installs on the website with instructions for the user)

**Reply by ajwfrost** [reply]
@ascentcreative I was just about to suggest, unless this is for offline use then wouldn't it be simpler to link to the runtime files from the website. I know of a few companies who do that as the steps for installation.. then the end user just gets directly covered by the license and there's no redistribution of the software happening.

For the sidecar distribution options, can you email me direct please, andrew.frost at harman.com

thanks

**Reply by kieranmetcalfe** [reply]
Cheers - yeah, that might be the easiest option, although having a few not-too-tech-savvy users, we were hoping to minimise the steps needed if possible. I'll drop you an email about the sidecar distribution agreement, just so I can have all the info to present to the owners.

thanks again
